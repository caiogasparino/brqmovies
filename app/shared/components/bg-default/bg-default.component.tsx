import { ReactNode } from "react";

import { Container } from "./bg-default.styles";

export type BgDefaultProps = {
    children: ReactNode;
};

const BgDefault: React.FC<BgDefaultProps> = ({ children }) => {
    return <Container testID="bg-default">{children}</Container>;
};

export { BgDefault };
