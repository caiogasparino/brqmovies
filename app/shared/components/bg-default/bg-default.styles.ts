import { SafeAreaView } from "react-native-safe-area-context";
import styled from "styled-components/native";

import { BgDefaultProps } from "./bg-default.component";

export const Container = styled(SafeAreaView)<BgDefaultProps>`
    flex: 1;
    padding-bottom: ${({ theme }) => theme.METRICS?.PADDING.x24}px;
    background-color: ${({ theme }) => theme.COLORS?.BRAND[100]};
`;
