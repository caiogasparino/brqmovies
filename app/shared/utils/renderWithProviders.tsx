import { render, RenderAPI } from "@testing-library/react-native";
import React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

import { NavigationContainer } from "@react-navigation/native";
import { THEME_LIGHT } from "app/shared/design/themes/theme-light";
import { ThemeProvider } from "styled-components";

const renderWithProviders = (component: React.ReactElement): RenderAPI => {
    return render(
        <NavigationContainer>
            <SafeAreaProvider>
                <ThemeProvider theme={THEME_LIGHT}>{component}</ThemeProvider>
            </SafeAreaProvider>
        </NavigationContainer>,
    );
};

export { renderWithProviders };
