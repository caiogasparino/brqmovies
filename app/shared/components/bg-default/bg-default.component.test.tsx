import { renderWithProviders } from "app/shared/utils/renderWithProviders";
import { BgDefault, BgDefaultProps } from "./bg-default.component";

const defaultProps: BgDefaultProps = {
    children: "Test Children",
};

describe("BgDefault", () => {
    it("should match snapshot", () => {
        renderWithProviders(<BgDefault {...defaultProps} />);
        expect(defaultProps).toMatchSnapshot();
    });

    it("should render", () => {
        renderWithProviders(<BgDefault {...defaultProps} />);
        expect(true).toBeTruthy();
    });
});
