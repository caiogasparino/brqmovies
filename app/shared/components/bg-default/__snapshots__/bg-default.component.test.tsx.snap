// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BgDefault should match snapshot 1`] = `
{
  "children": "Test Children",
}
`;
